### PR TITLE
[PM-14033] Configure test workflow simulator type and iOS version with config files

### PR DESCRIPTION
**Pull request by KatherineInCode**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-14033

## 📔 Objective

Similar to how we now have an `.xcode-version` file to denote which version of Xcode should be used when building/testing, this creates `.test-simulator-device-name` and `.test-simulator-ios-version` files to denote which device tests should be run against and which iOS version on that device, respectively.

I went back and forth on whether to make this two files like this, or one file with the full `name=[blah],OS=[blah]` string; I went with the former because it's more straightforward, and easier to potentially use in other workflows/tools in the future.

It would be nice if there were a way to use this in [BitwardenTestCase](https://github.com/bitwarden/ios/blob/main/GlobalTestHelpers/Support/BitwardenTestCase.swift#L12) as well, but I don't think that's readily possible without pre-build cleverness unfortunately.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -72,6 +72,8 @@ jobs:
       - name: Read default Xcode version
         run: |
           echo "DEFAULT_XCODE_VERSION=$(cat .xcode-version | tr -d '\n')" >> "$GITHUB_ENV"
+          echo "DEFAULT_SIMULATOR_NAME=$(cat .test-simulator-device-name | tr -d '\n')" >> "$GITHUB_ENV"
+          echo "DEFAULT_SIMULATOR_VERSION=$(cat .test-simulator-ios-version | tr -d '\n')" >> "$GITHUB_ENV"
```
Comment (fedemkr) [inline .github/workflows/test.yml:76]: 🤔 Maybe the name of the step should be updated given that it does more than just the Xcode version. What do you think?

**Review by fedemkr — COMMENTED** [review]
🤔 Looks good! About the configuration being in different files I was thinking that most likely these files will be updated alongside when the Xcode version changes so IMO we could have everything on one file and use the `source` / `.` command to load if the file looks something like this:
```
XCODE_VERSION=15.4
SIMULATOR_NAME=iPhone 15 Pro
SIMULATOR_VERSION=18.0
```
So we just do `. /path/to/theconfigfile` and the variables will be automatically available and ready for use in the step to be added to `$GITHUB_ENV` without needing to do any `cat` or `sed` commands.
Thoughts?

File: .github/workflows/test.yml (hunk 1 above)
Comment (KatherineInCode) [reply .github/workflows/test.yml:76]: 🤦🏼‍♀️ It is in the Xcode 16 branch, I just didn't bring it over here

**Comment by KatherineInCode** [comment]
> 🤔 Looks good! About the configuration being in different files I was thinking that most likely these files will be updated alongside when the Xcode version changes so IMO we could have everything on one file and use the `source` / `.` command to load if the file looks something like this:

I considered that option, too, and went down a long rabbit hole of how to import environment variables from a file into GitHub actions.

The thing is, the `.xcode-version` file is [something of a standard](https://github.com/fastlane/ci/blob/master/docs/xcode-version.md), and used by other things in the ecosystem (particularly Fastlane). So there's benefit to having that as a standalone file itself, which is why ultimately I [did it that way](https://github.com/bitwarden/ios/pull/1029).

Ultimately it's probably six of one, half dozen of another. If I had a good reason for abandoning the `.xcode-version` standard, then yeah, I'd be inclined to put them all in one file.

**Review by fedemkr — APPROVED** [review]
Agree makes total sense, thanks for explaining @KatherineInCode ! 🎉

**Pull request merged**